Vanilla Compatibility and Phosphor Changes

Spearkiller opened the pull request:
This PR was created based off the following suggestion: https://github.com/thedarkcolour/ForestryCE/issues/169

~~- The Lush line of bees now has a special comb type: Luminous~~
  ~~- Luminous Comb can be centrifuged into beeswax, honey drops (70% chance) and phosphorus (60% chance)~~
  - Phosphorus is now the specialty product of the ~~Lush line of bees~~ Luxurious Bee
- Simmering Combs now produce refractory wax, and a new type of propolis, Volcanic Propolis.
  - This replaces phosphorus which would previously be obtained when processing simmering combs.
  - Volcanic Propolis is produced 70% of the time.
  - Volcanic Propolis can be placed in a squeezer alongside dirt, sand, cobblestone and magma blocks to produce lava, replacing the previous uses for phosphorus.
    - The material placed inside the squeezer determines the amount of lava produced, with magma blocks producing the most.
- ~~Removed Honey Pots.~~
- Added some new recipes to interact with vanilla and give use to some lesser used items:
  - 4 Ice Shards can be crafted into a block of ice
  - 1 Phosphor and 1 stick can be crafted into 6 torches
  - 1 beeswax and 1 string can be crafted into a candle
  - 1 bottle and 2 honey drops can be crafted into a honey bottle. This replaces the old Honey Pot item and recipe.
    - This does provide a roundabout way of turning honey into sugar as vanilla minecraft allows you to craft a honey bottle into 3 sugar. 
  - 8 honey drops can be crafted into a block of honey. ~~This process is reversible.~~
  - A block of honey can be placed in a squeezer to produce 800mb of honey.
    - This is the same as putting 8 drops of honey in individually, but it processes it 25% faster.
    - To counter this, no propolis will be produced as a byproduct.
  - Honeycomb (from vanilla minecraft) can be placed in a centrifuge to produce 1 beeswax.

Note that I had wanted phosphorus to be able to be used as an alternative to glowing ink sacs but I didn't know how to implement that.

Spearkiller committed:
Added Luminous Comb and Volcanic Propolis

Spearkiller committed:
Updated recipes involving phosphor to use volcanic propolis instead

Also changed yields, speeds, and added an extra recipe using magma

Spearkiller committed:
Added a recipe to turn Ice Shards into ice

Spearkiller committed:
Added recipes for phosphor torches and beeswax candles

Spearkiller committed:
Added a recipe to turn vanilla combs into beeswax

Spearkiller committed:
Removed Honey Pot, added Vanilla Honey Compat

two honey drops makes one honey bottle
as such, a honey block can be turned into 8 honey drops
it can then also be squeezed into 8x as much honey as a single drop (though i removed the propolis chance in place of a higher processing speed)

Spearkiller committed:
fixed a miscalculation in previous commit

Spearkiller committed:
run datagen for new assets

Spearkiller committed:
Added models that weren't datagened

Spearkiller committed:
Changed colouring of Luminous and Simmering combs

Spearkiller committed:
fixed new recipes overwriting vanilla ones

Spearkiller committed:
made candles 50% cheaper

Spearkiller committed:
made phosphorous less plentiful

Spearkiller committed:
Lush line of bees now actually make Luminous Comb

Spearkiller committed:
Phosphor can now be used to make Glowing Item Frames

thedarkcolour commented on src/main/java/forestry/apiculture/features/ApicultureItems.java:
Unfortunately, we can't remove items from Forestry in the middle of a major Minecraft version. For the time being, we can remove its recipe and hide it from the creative tab, and make an Issue suggesting to remove it in 1.21.1.

Spearkiller replied:
https://github.com/thedarkcolour/ForestryCE/issues/169 suggests it to be removed but will be closed when this PR is accepted. I will make a new issue suggesting its removal in 1.21.
Resolved in https://github.com/thedarkcolour/ForestryCE/commit/4ae31d2a1910db04f2d5cdc4a2266e9e4ed71911

thedarkcolour commented on src/main/java/forestry/core/data/ForestryItemModels.java:
See above comment about not removing items mid-MC version

thedarkcolour commented on src/main/java/forestry/core/data/ForestryRecipeProvider.java:
```diff
@@ -490,21 +490,16 @@ private static void registerFoodRecipes(MKRecipeProvider recipes) {
 			recipe.pattern("###");
 		});
 
-		recipes.shapedCrafting(RecipeCategory.FOOD, ApicultureItems.HONEY_POT, recipe -> {
```
Even though we can't remove the honey pot item till 1.21, we can still remove its recipe.

Spearkiller replied:
Resolved in https://github.com/thedarkcolour/ForestryCE/commit/4ae31d2a1910db04f2d5cdc4a2266e9e4ed71911

thedarkcolour commented on src/main/java/forestry/core/data/ForestryRecipeProvider.java:
```diff
@@ -773,6 +768,38 @@ private static void registerCoreRecipes(MKRecipeProvider recipes) {
 
+		recipes.shapedCrafting("compressed_ice_shards", RecipeCategory.MISC, Items.ICE, 1, recipe -> {
+			recipe.define('#', CoreItems.CRAFTING_MATERIALS.get(EnumCraftingMaterial.ICE_SHARD));
+			recipe.pattern("## ");
```
I assume the spaces at the end of each line are not intentional? It's perfectly fine to do "##" without the extra space

Spearkiller replied:
Yeah that was unintentional. I'm not 100% sure why I did it that way tbh
Resolved in https://github.com/thedarkcolour/ForestryCE/commit/b41e77d112f63389c08de3076a428fb65c60dae3

thedarkcolour commented on src/main/java/forestry/core/data/ForestryRecipeProvider.java:
```diff
@@ -773,6 +768,38 @@ private static void registerCoreRecipes(MKRecipeProvider recipes) {
+			recipe.pattern("VVV");
+		});
+
+		recipes.shapelessCrafting(RecipeCategory.MISC, ApicultureItems.HONEY_DROP, 8, Items.HONEY_BLOCK);
```
This recipe (Honey Block -> Honey Drop) might be a little overpowered for getting honey drops.

Spearkiller replied:
I suppose it makes enough sense to have it be a one-way recipe.
Removed and resolved in https://github.com/thedarkcolour/ForestryCE/commit/b41e77d112f63389c08de3076a428fb65c60dae3

thedarkcolour commented on src/main/java/forestry/core/data/ForestryRecipeProvider.java:
```diff
@@ -773,6 +768,38 @@ private static void registerCoreRecipes(MKRecipeProvider recipes) {
+
+		recipes.shapedCrafting("phosphor_torches", RecipeCategory.MISC, Items.TORCH, 6, recipe -> {
+			recipe.define('P', CoreItems.CRAFTING_MATERIALS.get(EnumCraftingMaterial.PHOSPHOR));
+			recipe.define('|', Items.STICK);
```
Use `Tags.Items.RODS_WOODEN` instead of `Items.STICK`.

Spearkiller replied:
Resolved in https://github.com/thedarkcolour/ForestryCE/commit/b41e77d112f63389c08de3076a428fb65c60dae3

thedarkcolour commented on src/main/java/forestry/core/data/ForestryRecipeProvider.java:
```diff
@@ -773,6 +768,38 @@ private static void registerCoreRecipes(MKRecipeProvider recipes) {
+		});
+
+		recipes.shapedCrafting("beeswax_candles", RecipeCategory.MISC, Items.CANDLE, 1, recipe -> {
+			recipe.define('|', Items.STRING);
```
Use `Tags.Items.STRING` instead of `Items.STRING`.

Spearkiller replied:
Resolved in https://github.com/thedarkcolour/ForestryCE/commit/b41e77d112f63389c08de3076a428fb65c60dae3

thedarkcolour commented on src/main/java/forestry/core/data/ForestryRecipeProvider.java:
```diff
@@ -1958,29 +2002,42 @@ private static void registerSqueezer(Consumer<FinishedRecipe> consumer) {
+					ingredients.add(Ingredient.of(Items.COBBLESTONE));
 				}))
-				.setFluidOutput(new FluidStack(Fluids.LAVA, 1600))
+				.setFluidOutput(new FluidStack(Fluids.LAVA, 750))
```
I think 500 mB is okay, and it would round better (2 * 500 mB = 1 B)

Spearkiller replied:
Resolved in https://github.com/thedarkcolour/ForestryCE/commit/b41e77d112f63389c08de3076a428fb65c60dae3

I did change it a bit further
- Sand recipes take 20 seconds and yield 500mb
- Cobble recipe now takes 30 seconds to *also* yield 500mb. This is because a lot of mods add ways to turn cobble into sand and I didn't like the disparity. Since sand is (often) a 1:1 conversion, but with extra steps, to me it makes sense that the only difference ought to be in processing times. Plus cobble is easier to get than sand. It just makes sense to me, I suppose. Willing to change it again if need be.
- Magma Block recipe takes 20 seconds but now yields only 1000mb. I think 1500mb was just a bit too much in retrospect, and was only that high because the previous recipes were too. If we're downsizing, may as well do so here too.

thedarkcolour commented on src/main/resources/assets/forestry/textures/item/honey_pot.png:
This has to stay in, unfortunately.

Spearkiller replied:
Resolved in https://github.com/thedarkcolour/ForestryCE/commit/4ae31d2a1910db04f2d5cdc4a2266e9e4ed71911

thedarkcolour reviewed (changes requested):
Love the ideas. Please read all my comments!

thedarkcolour commented:
This PR will close the following issue: #169

Spearkiller committed:
Soft revert of Honey Pot removal

Recipe is still removed, and creative tab entry is also still removed.

Spearkiller committed:
Update ForestryRecipeProvider.java

Removed spaced on the end of Ice Shards -> Ice
Fixed Phosphor Candles and Torches not using tags
Removed Honey Block -> Honey Drops
Rebalanced Volcanic Propolis squeezer recipes (Reduced lava yields, increased processing time for cobblestone recipe.

Spearkiller committed:
Removed Luminous Comb as per recommendations

Phosphor can also no longer be used for glow frames
Phosphor is now a specialty product of luxuriant bees

Spearkiller committed:
Changed phosphorusfrom a specialty to a product

Spearkiller committed:
Merge branch '1.20.1' into phosphorescent-update

Spearkiller committed:
Removed String -> Silk centrifuge recipe

Spearkiller committed:
Reduced cost for Woven Silk

Spearkiller committed:
forgot to add file to commit
